### PR TITLE
Added request validation

### PR DESCRIPTION
**Pull request by Yukio0315**
I added all the request validation. Please check it whether it is correct or not.

I also bug fixed logout error handling. If invalid token is sent, panic happened. Because of `amountOfdeleted = 0 and err = nil` but NewError was called.

### CODE REVIEW

File: app/interfaces/controllers/error.go [hunk 1]
```diff
@@ -9,3 +9,9 @@ func NewError(err error) *Error {
 		Message: err.Error(),
 	}
 }
+
+func ValidationError(info string, err error) *Error {
+	return &Error{
+		Message: "ValidationError about " + info + ":" + err.Error(),
```
Comment (FuyuByakko) [inline app/interfaces/controllers/error.go]: change about to "for".
And since this is a message already, add a space "Validation Error".

**Review by FuyuByakko — COMMENTED** [review]
1 tiny comment, otherwise looks awesome!

File: app/interfaces/controllers/error.go (hunk 1 above)
Comment (Yukio0315) [reply app/interfaces/controllers/error.go]: Thanks! I changed message.

**Review by FuyuByakko — APPROVED** [review]
LGTM!!

**Pull request merged**